cmd/syncthing/main.go

```diff
@@ -207,132 +207,145 @@ var (
 	guiAddress     string
 	guiAPIKey      string
 	generateDir    string
-	noRestart      = os.Getenv("STNORESTART") != ""
-	noUpgrade      = os.Getenv("STNOUPGRADE") != ""
-	profiler       = os.Getenv("STPROFILER")
-	guiAssets      = os.Getenv("STGUIASSETS")
-	cpuProfile     = os.Getenv("STCPUPROFILE") != ""
-	stRestarting   = os.Getenv("STRESTART") != ""
-	innerProcess   = os.Getenv("STNORESTART") != "" || os.Getenv("STMONITORED") != ""
-)
+	noRestart      bool
+	profiler       string
+	assetDir       string
+	cpuProfile     bool
+	stRestarting   bool
+	logFlags       int
+}
+
+func defaultRuntimeOptions() RuntimeOptions {
+	options := RuntimeOptions{
+		noRestart:    os.Getenv("STNORESTART") != "",
+		profiler:     os.Getenv("STPROFILER"),
+		assetDir:     os.Getenv("STGUIASSETS"),
+		cpuProfile:   os.Getenv("STCPUPROFILE") != "",
+		stRestarting: os.Getenv("STRESTART") != "",
+		logFile:      "-", // Output to stdout
+		logFlags:     log.Ltime,
+	}
+
+	if options.assetDir != "" {
+		options.assetDir = locations[locGUIAssets]
+	}
+
+	if os.Getenv("STTRACE") != "" {
+		options.logFlags = log.Ltime | log.Ldate | log.Lmicroseconds | log.Lshortfile
+	}
 
-func parseCommandLineOptions() {
 	if runtime.GOOS == "windows" {
-		// On Windows, we use a log file by default. Setting the -logfile flag
-		// to "-" disables this behavior.
-		flag.StringVar(&logFile, "logfile", "", "Log file name (use \"-\" for stdout)")
+		options.logFile = locations[locLogFile]
```

Comment:
It seemed to me the previous behavior was:

* if it's windows, defaults to blank;
* if it's not windows, defaults to "-";
* inside `main()`, if blank change it to `locations[locLogFile]`;

I think the behavior is the same, just one step shorter:
* defaults to "-"
* if it's windows, defaults to `locations[locLogFile]`

or did I miss something?